src/state/block/hooks.ts

```diff
@@ -8,6 +9,35 @@ import { State } from '../types'
 
 const REFRESH_BLOCK_INTERVAL = 6000
 
+export const useFetchBlockNumber = () => {
+  const [isLoading, setFetch] = useState(false)
+
+  const dispatch = useAppDispatch()
+
+  const fetchBlockNumber = useCallback(() => {
+    if (!isLoading) {
+      setFetch(true)
+    }
+  }, [isLoading])
+
+  const { isValidating, data: blockNumber } = useSWR(isLoading ? ['blockNumber'] : null, () =>
```

Comment:
How about use `mutate`, and `const isLoading = useDebounce(isValidating)`  to get the delay loading?